### DISCUSSION TITLE
`node-magickwand` - ImageMagick for Node.js has reached beta status

### DISCUSSION THREAD

**Discussion by mmomtchev** [ImageMagick/ImageMagick#6657] [General]
I am pleased to announce that `node-magickwand` - the port of the ImageMagick library to Node.js has reached beta status and it is now feature-complete.

It offers native Javascript bindings for the full Magick++ and MagickWand APIs and select parts of the MagickCore API. It is fully integrated with `TypedArray`  and `Buffer`, supports `Promise`-based asynchronous operations, including multiple concurrent calls and it has full TypeScript support.

It is designed to work with Express.js-like middleware and it should never block the event loop when only the asynchronous methods are used.

https://www.npmjs.com/package/node-magickwand

Get it from npm:
```
npm i node-magickwand
```

It is the showcase project of SWIG Node-API.

**Comment by urban-warrior** [comment]
Fantastic news.  Thanks to the team that delivered the Node.JS wrapper.

**Reply by mmomtchev** [reply]
I am alone on this project.  [edited by moderator to meet content requirements]

If you can post a link to this library somewhere  [edited by moderator to meet content requirements]

**Comment by urban-warrior** [comment]
We've already added a link to https://imagemagick.org/script/develop.php.

**Reply by mmomtchev** [reply]
It is much appreciated.

**Comment by mmomtchev** [comment]
@urban-warrior I am very happy to present you the first successful WASM build of `node-magickwand` (which will probably get a new name now that is more than simply `node-something`) - that works both in Node.js and in the browser.

The first beta version is a matter of a few weeks. The big thing missing at the moment is async mode.

https://garga.net/imagemagick-wasm.zip

**Comment by urban-warrior** [comment]
We appreciate your contribution to the ImageMagick ecosystem. If you're part of the Node.js user community, we encourage you to give it a try and don't hesitate to share any feedback or report any issues you encounter here.
